### PR TITLE
BUG: fix KeyError with list of a single, missing, element

### LINKED ISSUE #27148: Indexing single level with list of missing labels raises if (and only if) it has length 1
#### Code Sample, a copy-pastable example if possible

```python
In [2]: mi = pd.MultiIndex.from_tuples([[1, 2, -1], [3,4, -2], [5, 6, -3]])                                                                                                                                                                   

In [3]: s = pd.Series(1, mi)                                                                                                                                                                                                                  

In [4]: s.loc[[100, 101]]                                                                                                                                                                                                                     
Out[4]: Series([], dtype: int64)

In [5]: s.loc[[100]]                                                                                                                                                                                                                          
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-5-7a5bc11c892d> in <module>
----> 1 s.loc[[100]]

~/nobackup/repo/pandas/pandas/core/indexing.py in __getitem__(self, key)
   1427 
   1428             maybe_callable = com.apply_if_callable(key, self.obj)
-> 1429             return self._getitem_axis(maybe_callable, axis=axis)
   1430 
   1431     def _is_scalar_access(self, key):

~/nobackup/repo/pandas/pandas/core/indexing.py in _getitem_axis(self, key, axis)
   1829                     raise ValueError('Cannot index with multidimensional key')
   1830 
-> 1831                 return self._getitem_iterable(key, axis=axis)
   1832 
   1833             # nested tuple slicing

~/nobackup/repo/pandas/pandas/core/indexing.py in _getitem_iterable(self, key, axis)
   1131             # A collection of keys
   1132             keyarr, indexer = self._get_listlike_indexer(key, axis,
-> 1133                                                          raise_missing=False)
   1134             return self.obj._reindex_with_indexers({axis: [keyarr, indexer]},
   1135                                                    copy=True, allow_dups=True)

~/nobackup/repo/pandas/pandas/core/indexing.py in _get_listlike_indexer(self, key, axis, raise_missing)
   1068         # if it cannot handle:
   1069         indexer, keyarr = ax._convert_listlike_indexer(key,
-> 1070                                                        kind=self.name)
   1071         # We only act on all found values:
   1072         if indexer is not None and (indexer != -1).all():

~/nobackup/repo/pandas/pandas/core/indexes/multi.py in _convert_listlike_indexer(self, keyarr, kind)
   2255             mask = check == -1
   2256             if mask.any():
-> 2257                 raise KeyError('%s not in index' % keyarr[mask])
   2258 
   2259         return indexer, keyarr

KeyError: '[100] not in index'
```

#### Problem description

Because of a [trivial typo](https://github.com/pandas-dev/pandas/blob/af7f2ef73e449f01acc6de47463c9b1440c6b0fb/pandas/core/indexing.py#L1821), behavior depends on whether the list has length at least 2. The "one item" behavior is arguably seldomly used, and future incompatible (xref: #20916 ).

#### Expected Output

Like ``Out[4]:``.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : b4913295ab549652d1714082a60ec730436ed612
python           : 3.7.3.candidate.1
python-bits      : 64
OS               : Linux
OS-release       : 4.9.0-9-amd64
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : it_IT.UTF-8
LOCALE           : it_IT.UTF-8

pandas           : 0.25.0.dev0+849.gb4913295a
numpy            : 1.16.4
pytz             : 2016.7
dateutil         : 2.8.0
pip              : 9.0.1
setuptools       : 41.0.1
Cython           : 0.29.2
pytest           : 4.6.3
hypothesis       : 3.71.11
sphinx           : 1.4.9
blosc            : None
feather          : None
xlsxwriter       : 0.9.6
lxml.etree       : 4.3.2
html5lib         : 0.999999999
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: 0.2.1
bs4              : 4.5.3
bottleneck       : 1.2.1
fastparquet      : None
gcsfs            : None
lxml.etree       : 4.3.2
matplotlib       : 3.0.2
numexpr          : 2.6.9
openpyxl         : 2.3.0
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.1.0
sqlalchemy       : 1.0.15
tables           : 3.4.4
xarray           : None
xlrd             : 1.0.0
xlwt             : 1.3.0
xlsxwriter       : 0.9.6


</details>

### PR DESCRIPTION
**Pull request by toobaz**
- [x] closes #27148
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: doc/source/whatsnew/v0.25.0.rst [hunk 1]
```diff
@@ -768,6 +768,7 @@ Indexing
 - Bug in which :meth:`DataFrame.to_csv` caused a segfault for a reindexed data frame, when the indices were single-level :class:`MultiIndex` (:issue:`26303`).
 - Fixed bug where assigning a :class:`arrays.PandasArray` to a :class:`pandas.core.frame.DataFrame` would raise error (:issue:`26390`)
 - Allow keyword arguments for callable local reference used in the :meth:`DataFrame.query` string (:issue:`26426`)
+- Fixed a ``KeyError`` when indexing a :class:`MultiIndex`` level with a list containing exactly one label, which is missing (:issue:`16396`)
```
Comment (simonjayhawkins) [inline doc/source/whatsnew/v0.25.0.rst]: should the issue 27148 be mentioned?

File: pandas/core/indexing.py [hunk 2]
```diff
@@ -1818,7 +1818,7 @@ def _getitem_axis(self, key, axis=None):
                                               "multidimensional key is not "
                                               "implemented")
 
-                if (not isinstance(key, tuple) and len(key) > 1 and
+                if (not isinstance(key, tuple) and len(key) > 0 and
```
Comment (simonjayhawkins) [inline pandas/core/indexing.py]: do we need the greater than here?

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 3]
```diff
@@ -123,6 +123,17 @@ def test_loc_multiindex_missing_label_raises(self):
         with pytest.raises(KeyError, match=r"^2$"):
             df.loc[2]
 
+        # Lists with missing labels do not raise:
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/multiindex/test_loc.py]: these should probably be separate tests, although it does mean duplicating dataframe setup.

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 4]
```diff
@@ -123,6 +123,17 @@ def test_loc_multiindex_missing_label_raises(self):
...
+        expected = df.iloc[[0, 1]]
+        result = df.loc[[2, 4]]
+        tm.assert_frame_equal(result, expected)
+
+        # ... even when none is found:
+        expected = df.iloc[[]]
+        for l in [2], [2, 3]:  # GH 27148
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/multiindex/test_loc.py]: better to use parametrisation than a for loop.

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 5]
```diff
@@ -123,6 +123,17 @@ def test_loc_multiindex_missing_label_raises(self):
...
+            result = df.loc[[2]]
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/multiindex/test_loc.py]: should this be [2]?

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 6]
```diff
@@ -264,15 +275,16 @@ def convert_nested_indexer(indexer_type, keys):
...
 
 
-@pytest.mark.parametrize('indexer, is_level1, expected_error', [
-    ([], False, None),  # empty ok
-    (['A'], False, None),
-    (['A', 'D'], False, None),
-    (['D'], False, r"\['D'\] not in index"),  # not any values found
-    (pd.IndexSlice[:, ['foo']], True, None),
-    (pd.IndexSlice[:, ['foo', 'bah']], True, None)
+@pytest.mark.parametrize('indexer, level, expected_error', [
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/multiindex/test_loc.py]: you can now drop the expected_error and remove the relevant code.

**Review by simonjayhawkins — COMMENTED** [review]
generally lgtm. a few minor queries.

File: doc/source/whatsnew/v0.25.0.rst (hunk 1 above)
Comment (toobaz) [reply doc/source/whatsnew/v0.25.0.rst]: thanks!

File: pandas/core/indexing.py (hunk 2 above)
Comment (toobaz) [reply pandas/core/indexing.py]: removed

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 5 above)
Comment (toobaz) [reply pandas/tests/indexing/multiindex/test_loc.py]: ouch... fixed

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 6 above)
Comment (toobaz) [reply pandas/tests/indexing/multiindex/test_loc.py]: indeed!

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 3 above)
Comment (toobaz) [reply pandas/tests/indexing/multiindex/test_loc.py]: I think that's a manageable test... and not worth a fixture (if that's what you mean)

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 4 above)
Comment (toobaz) [reply pandas/tests/indexing/multiindex/test_loc.py]: (related to above)

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 7]
```diff
@@ -283,22 +294,18 @@ def test_loc_getitem_duplicates_multiindex_missing_indexers(indexer, is_level1,
...
         exp_idx = MultiIndex.from_product([['A'], ['foo', 'bar', 'baz']],
                                           names=['one', 'two'])
         expected = Series(np.arange(3, dtype='int64'),
                           index=exp_idx).sort_index()
 
-    if expected_error is not None:
-        with pytest.raises(KeyError, match=expected_error):
-            s.loc[indexer]
-    else:
-        result = s.loc[indexer]
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/multiindex/test_loc.py]: this line and the next should remain.

Comment (toobaz) [reply pandas/tests/indexing/multiindex/test_loc.py]: oh...

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 3 above)
Comment (toobaz) [reply pandas/tests/indexing/multiindex/test_loc.py]: No, sorry, I misread "it does mean" in "it doesn't mean". Not a fan of duplicating the setup to be honest.

File: pandas/tests/indexing/multiindex/test_loc.py [hunk 8]
```diff
@@ -283,21 +294,20 @@ def test_loc_getitem_duplicates_multiindex_missing_indexers(indexer, is_level1,
 
     if indexer == []:
         expected = s.iloc[[]]
-    elif is_level1:
+    elif level == 1:
         expected = Series([0, 3, 6], index=MultiIndex.from_product(
             [['A', 'B', 'C'], ['foo']], names=['one', 'two'])).sort_index()
+    elif level == -1:
+        # Empty: use idx[:0] as index to have appropriate (hidden) labels
+        expected = Series([], index=idx[:0], dtype='int64')
```
Comment (simonjayhawkins) [inline pandas/tests/indexing/multiindex/test_loc.py]: i think this is the same as the `expected = s.iloc[[]]` above.

with the exception case now removed, I think that this whole test could be simplified..

```python
@pytest.mark.parametrize('indexer, expected', [
    ([], []),  # empty ok
    (['A'], slice(3)),
    (['A', 'D'], slice(3)),
    (['D', 'E'], []),  # no values found - fine
    (['D'], []),  # same, with single item list: GH 27148
    (pd.IndexSlice[:, ['foo']], slice(2, None, 3)),
    (pd.IndexSlice[:, ['foo', 'bah']], slice(2, None, 3))
])
def test_loc_getitem_duplicates_multiindex_missing_indexers(indexer, expected):
    # GH 7866
    # multi-index slicing with missing indexers
    idx = MultiIndex.from_product([['A', 'B', 'C'],
                                   ['foo', 'bar', 'baz']],
                                  names=['one', 'two'])
    s = Series(np.arange(9, dtype='int64'), index=idx).sort_index()
    expected = s.iloc[expected]
    result = s.loc[indexer]
    tm.assert_series_equal(result, expected)
```

Comment (toobaz) [reply pandas/tests/indexing/multiindex/test_loc.py]: Right, thanks! I only changed the name of the second parameter to "pos".

File: pandas/tests/indexing/multiindex/test_loc.py (hunk 4 above)
Comment (jreback) [reply pandas/tests/indexing/multiindex/test_loc.py]: I agree here, would be nice to parameterize this, easier to grok; no big deal to copy-paste the setup (no fixture really needed here)

**Review by simonjayhawkins — APPROVED** [review]
@toobaz lgtm. @jreback

**Comment by jreback** [comment]
thanks @toobaz

**Pull request merged**